src/aa_remove_data/pb_utils.py

```diff
@@ -0,0 +1,164 @@
+        """Convert the name of a sample type to CamelCase to correspond to
+        that sample type's class name.
+
+        Args:
+            sample_type (str): Name of sample type.
+
+        Returns:
+            str: Name of sample class, e.g VectorDouble.
+        """
+        # Split the enum name by underscores and capitalize each part
+        parts = sample_type.split("_")
+        return "".join(part.capitalize() for part in parts)
+
+    def convert_to_datetime(self, year: int, seconds: int) -> datetime:
+        """Get the date and time from a year and the number of seconds passed.
+        Args:
+            year (int): A year
+            seconds (int): The number of seconds into that year that have
+            passed.
+        Returns:
+            datetime: A datetime object of the correct date and time.
+        """
+        return datetime(year, 1, 1) + timedelta(seconds=seconds)
+
+    def get_datastr(self, sample: type, year: int) -> str:
+        """Get a string contaiing information about a sample.
+        Args:
+            sample (type): A sample from a pb file.
+            year (int): The year the sample was collected.
+        Returns:
+            str: A string containing the sample information.
+        """
+        date = self.convert_to_datetime(year, sample.secondsintoyear)
+        return (
+            f"{date}    {sample.secondsintoyear:8d}    {sample.nano:9d}"
+            f"    {sample.val}\n"
+        )
+
+    def get_sample_type(self) -> str:
+        """Get the name of a pb file's sample type using information in its
+        header.
+
+        Returns:
+            str: Name of sample type, e.g VECTOR_DOUBLE.
+        """
+        type_descriptor = self.header.DESCRIPTOR.fields_by_name["type"]
+        enum_descriptor = type_descriptor.enum_type
+        return enum_descriptor.values_by_number[self.header.type].name
+
+    def get_sample_class(self) -> type:
+        """Get the EPICSEvent_pb2 class corresponding to samples in a pb file.
+        Instances of this class can interpret pb samples of a matching type.
+
+        Returns:
+            type: pb sample class.
+        """
+        # Ensure self.sample_type is set first.
+        if not self.sample_type:
+            self.sample_type = self.get_sample_type()
+        sample_type_camel = self._convert_to_class_name(self.sample_type)
+        sample_class = getattr(EPICSEvent_pb2, sample_type_camel)
+        return sample_class
+
+    def write_to_txt(self, filepath: PathLike):
+        """Write a text file from a PBUtils object.
+
+        Args:
+            filepath (PathLike): Filepath for file to be written.
+        """
+        pvname = self.header.pvname
+        year = self.header.year
+        sample_type = self.get_sample_type()
+        data_strs = [self.get_datastr(sample, year) for sample in self.samples]
+        with open(filepath, "w") as f:
+            f.write(f"{pvname}, {sample_type}, {year}\n")
```

Comment:
I think this is writing the header and then the body using `writelines`. Could I have some comments here to explain what each of these writes correspond to?

Reply:
These have been added.

Reply:
Thanks